components/experimental/src/units/converter_factory.rs

```diff
@@ -38,20 +39,48 @@ impl From<Sign> for num_bigint::Sign {
     }
 }
 
-impl<'data> ConverterFactory<'data> {
-    #[cfg(feature = "datagen")]
-    pub fn from_payload(
-        payload: &'data UnitsInfoV1<'data>,
-        payload_store: &'data ZeroTrieSimpleAscii<ZeroVec<'data, u8>>,
-    ) -> Self {
-        Self {
-            payload,
-            payload_store,
-        }
+impl ConverterFactory {
+    icu_provider::gen_any_buffer_data_constructors!(
+        locale: skip,
+        options: skip,
+        error: ConversionError,
+        #[cfg(skip)]
```

Comment:
Nit: The macro should be used to add the docs and auto-generate `try_new`.

(You can ask @robertbastian for help on this)

Reply:
No problem, what should I actually do ?

Reply:
> what should I actually do ?

Use the form of the macro that generates all three functions `try_new`, `try_new_with_buffer_provider`, and `try_new_with_any_provider`. Currently you are using the form that generates only the last two.

Reply:
done!

Reply:
To generate the compiled data constructor:
```suggestion
    /// Creates a new [`ConverterFactory`] from compiled locale data.
    ///
    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
    ///
    /// [📚 Help choosing a constructor](icu_provider::constructors)
```